src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/CultureInfoAdapter.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Runtime.Serialization
+{
+    [DataContract(Name = "CultureInfo", Namespace = "http://schemas.datacontract.org/2004/07/System.Globalization")]
+    public struct CultureInfoAdapter
+    {
+        [DataMember(Name = "calendar", IsRequired = true)]
+        public string Calendar { get; set; }
+
```

Comment:
Yea, this doesn't work if the payload contains serialized calendar. I'm working on a solution for this.